addons/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/discovery/IsyRestDiscoveryService.java

```diff
@@ -123,6 +128,10 @@ protected void startScan() {
             logger.error("error in discover programs", e);
         }
 
+        // (TH) POTENTIAL HACK
+        // it would be nice to restart the web socket to the ISY after discovery is done
+        // the web socket is started to early, before any isy "things" are created, thus the initial ws update from the
+        // isy is missed
```

Comment:
TH, I have done quite a bit of work on a couple other bindings since I started this one.  There is a channelLinked method (I think in thing handler chain of hierarchy) which can be overridden and we can query the ISY for initial updates when items are discovered.  I spend a good bit of the year away from my house which has the isy so I have fallen behind on refactoring this binding until I get back there.  I will be returning to that house early jan so can get started soon.

Reply:
specifically, you can override channelLinked within the handler.  This method gets called whenever a channel is linked to an item, so then you get the info from the isy.

Reply:
One thing I noticed on the ISY is that the websocket (on initial connect) spits out device information that I cannot get on the REST interface. I'm not sure if that's intended or a bug on the ISY side. 

Just throwing this in: If the web service handler could keep the information from the initial contact (organized by insteon address, perhaps) and then forward that information as a update to the device on "channelLinked", that would be the best option I think (but probably also the hardest to implement)

Reply:
That’s interesting.  Never thought about all that other info maybe because I hadn’t got to wanting to use it.  Do you mean things like ramp rate, on level, etc for lights for example?  I haven’t used my isy now in 6 mos so a little rusty.

Reply:
One thing I noticed is a field called "UOM" (you may see some commented references in the code), which is only send on the initial websocket connect. I believe that's the current unit setting for the thermostat, which I should probably consider (need to read some more on the thermo doc's). I may have to do a different conversion based on that setting, but for now it's lost and I don't see it in the (browseable) REST interface. Not sure what else is sent that isn't part of what can be seen on the REST side